fladle-plugin/src/main/java/com/osacky/flank/gradle/FulladlePlugin.kt

```diff
@@ -166,3 +177,50 @@ fun writeAdditionalTestApps(strs: List<String>, flankGradleExtension: FlankGradl
     }
   )
 }
+
+val Project.isAndroidAppModule
+  get() = plugins.hasPlugin("com.android.application")
+val Project.isAndroidLibraryModule
+  get() = plugins.hasPlugin("com.android.library")
+
+// returns false if the module explicitly disabled testing or if it simply had no tests
+val Project.hasAndroidTest: Boolean
+  get() {
+    val fulladleModuleExtension = extensions.findByType(FulladleModuleExtension::class.java) ?: return false
+    if ((isAndroidLibraryModule || isAndroidAppModule) && !fulladleModuleExtension.enabled.get()) {
+      return false
+    }
+
+    val testedExtension = extensions.getByType<TestedExtension>()
+    var testsFound = true
+    testedExtension.testVariants.configureEach testVariant@{
+      if (!file("$projectDir/src/androidTest").exists()) {
+        println("Ignoring $name test variant in $path: No tests in $projectDir/src/androidTest")
+        testsFound = false
+      }
+      return@testVariant
+    }
+    return testsFound
+  }
+
+fun overrideRootLevelConfigs(flankGradleExtension: FlankGradleExtension, fulladleModuleExtension: FulladleModuleExtension) {
```

Comment:
this is overriding the local modules values, not the root values. can we rename?

Reply:
It is overriding root values. If a module is picked as a root and it specifies values in its `fulladleModuleConfig`, those are used to override the root values.